### ISSUE TITLE
PyPI version fails to build wheel under Python 3.8

### ISSUE THREAD

**Issue by wtbarnes** [PlasmaPy/PlasmaPy#737]
I am not able to install plasmapy from PyPI into a Python 3.8 environment created with conda. When doing so, I get the following compilation error which seems to originate from Cython:

```shell
    gcc -pthread -B /home/wbarnes/anaconda3/envs/plasmapy-test/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/wbarnes/anaconda3/envs/plasmapy-test/include/python3.8 -c plasmapy/_compiler.c -o build/temp.linux-x86_64-3.8/plasmapy/_compiler.o
    gcc -pthread -shared -B /home/wbarnes/anaconda3/envs/plasmapy-test/compiler_compat -L/home/wbarnes/anaconda3/envs/plasmapy-test/lib -Wl,-rpath=/home/wbarnes/anaconda3/envs/plasmapy-test/lib -Wl,--no-as-needed -Wl,--sysroot=/ build/temp.linux-x86_64-3.8/plasmapy/_compiler.o -o build/lib.linux-x86_64-3.8/plasmapy/_compiler.cpython-38-x86_64-linux-gnu.so
    building 'plasmapy.physics.parameters_cython' extension
    creating build/temp.linux-x86_64-3.8/plasmapy/physics
    gcc -pthread -B /home/wbarnes/anaconda3/envs/plasmapy-test/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/wbarnes/anaconda3/envs/plasmapy-test/lib/python3.8/site-packages/numpy/core/include -I/home/wbarnes/anaconda3/envs/plasmapy-test/include/python3.8 -c plasmapy/physics/parameters_cython.c -o build/temp.linux-x86_64-3.8/plasmapy/physics/parameters_cython.o -Wp,-w -Wno-unused-function
    plasmapy/physics/parameters_cython.c: In function ‘__Pyx_InitCachedConstants’:
    plasmapy/physics/parameters_cython.c:318:11: error: too many arguments to function ‘PyCode_New’
               PyCode_New(a, 0, k, l, s, f, code, c, n, v, fv, cell, fn, name, fline, lnos)
               ^
    plasmapy/physics/parameters_cython.c:1882:33: note: in expansion of macro ‘__Pyx_PyCode_New’
       __pyx_codeobj__5 = (PyObject*)__Pyx_PyCode_New(3, 0, 5, 0, CO_OPTIMIZED|CO_NEWLOCALS, __pyx_empty_bytes, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_tuple__4, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_kp_s_plasmapy_physics_parameters_cyth, __pyx_n_s_thermal_speed, 27, __pyx_empty_bytes); if (unlikely(!__pyx_codeobj__5)) __PYX_ERR(0, 27, __pyx_L1_error)
                                     ^~~~~~~~~~~~~~~~
    In file included from /home/wbarnes/anaconda3/envs/plasmapy-test/include/python3.8/compile.h:5:0,
                     from /home/wbarnes/anaconda3/envs/plasmapy-test/include/python3.8/Python.h:138,
                     from plasmapy/physics/parameters_cython.c:4:
    /home/wbarnes/anaconda3/envs/plasmapy-test/include/python3.8/code.h:122:28: note: declared here
     PyAPI_FUNC(PyCodeObject *) PyCode_New(
                                ^~~~~~~~~~
    plasmapy/physics/parameters_cython.c: In function ‘__Pyx_CreateCodeObjectForTraceback’:
    plasmapy/physics/parameters_cython.c:318:11: error: too many arguments to function ‘PyCode_New’
               PyCode_New(a, 0, k, l, s, f, code, c, n, v, fv, cell, fn, name, fline, lnos)
               ^
    plasmapy/physics/parameters_cython.c:3561:15: note: in expansion of macro ‘__Pyx_PyCode_New’
         py_code = __Pyx_PyCode_New(
                   ^~~~~~~~~~~~~~~~
    In file included from /home/wbarnes/anaconda3/envs/plasmapy-test/include/python3.8/compile.h:5:0,
                     from /home/wbarnes/anaconda3/envs/plasmapy-test/include/python3.8/Python.h:138,
                     from plasmapy/physics/parameters_cython.c:4:
    /home/wbarnes/anaconda3/envs/plasmapy-test/include/python3.8/code.h:122:28: note: declared here
     PyAPI_FUNC(PyCodeObject *) PyCode_New(
                                ^~~~~~~~~~
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/wbarnes/anaconda3/envs/plasmapy-test/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-4_qz5y_m/plasmapy/setup.py'"'"'; __file__='"'"'/tmp/pip-install-4_qz5y_m/plasmapy/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-xw_ol7li/install-record.txt --single-version-externally-managed --compile Check the logs for full command output.
```

This error actually pops up twice: once when it tries to build the wheel and then again when it tries to creat the build. Note that the version on PyPI says it is compatible with Python >= 3.6.

## Steps to reproduce

```shell
> conda create -n plasmapy-test python=3.8
> conda activate plasmapy-test
> pip install plasmapy
```

After switching to a 3.7 environment, I'm able to install via pip with no issue.

**Comment by StanczakDominik** [comment]
Right, that makes sense:

* The wheel probably needs building with a fresher, 3.8-compatible version of cython
* We weren't testing 3.8 during the last release (was it even out yet?)
* But we have since removed the file in question, and, in fact, Cython with it, with the intention to focus on numba more

Thus, if I'm correct, getting the current version out the door would actually fix this for free, essentially.

cc @Cadair: am I correct in thinking this bit of code I lovingly borrowed from SunPy would (somehow, automagically?) upload the wheel to PyPI? I imagine it would need some way to authenticate.

https://github.com/PlasmaPy/PlasmaPy/blob/46a77797ff64d61d4215f5b686687359f892d9dd/azure-pipelines.yml#L63-L83

That would get rid of a rather largeish chunk of the [release checklist](http://docs.plasmapy.org/en/latest/development/release_guide.html).

**Comment by wtbarnes** [comment]
Summon @nabobalis and his Azure pipelines knowledge

**Comment by wtbarnes** [comment]
(This is a moot point because it looks like it is already fixed on master, but it seems that pytest is a hard dependency though it is not picked up when installing via pip. It looks like this was because of the `from astropy.tests.helper import TestRunner` import in the now-defunct `_base_init.py`.

**Comment by nabobalis** [comment]
You would need that and to talk to someone who controls the azure account to add a specific auth on their web ui to upload the wheel.

**Comment by nabobalis** [comment]
See https://openastronomy-azure-pipelines.readthedocs.io/en/latest/publish.html

**Comment by StanczakDominik** [comment]
Thanks for the advice! The newly released 0.3 should fix this. Please take a look!
